lib/primer/view_components/linters/migrations/truncate_component.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Migrations
+      # Lint & autocorrect Truncate components
+      class TruncateComponent < RuboCop::Cop::Cop
+        INVALID_MESSAGE = <<~STR
+          `Primer::Truncate` is deprecated. Please use `Primer::Beta::Truncate` instead!
+        STR
+
+        def_node_matcher :truncate_component, <<~PATTERN
+          (send (const (const nil? :Primer) :Truncate) :new ...)
+        PATTERN
+
+        def_node_matcher :hash_with_inline_value?, <<~PATTERN
+          (hash ... (pair (sym :inline) (...)) ... )
+        PATTERN
+
+        def on_send(node)
+          return unless truncate_component(node)
+
+          add_offense(node, message: INVALID_MESSAGE)
+        end
+
+        def autocorrect(node)
+          return if hash_with_inline_value?(node.arguments.first)
+
+          lambda do |corrector|
+            corrector.replace(node.children.first, "Primer::Beta::Truncate")
```

Comment:
This won't be a totally safe transform, as `Primer::Truncate` used to return a `<div>` (which is `display: block`) where `Primer::Beta::Truncate` returns a `<span>` (which is `display: inline`). This could cause styling issues. Some options:

1. Try it and see if it works. There's 67 instances in gh/gh so not an overwhelming amount to check
2.  Rewrite to always include `tag: :div` in the arguments hash. Then we can look at writing a separate linter for `Primer::Beta::Truncate(tag: :div)`.

Reply:
That's a good point! I'll adjust to include the `tag`.